### PR TITLE
Add regex option for the include_tables and exclude_tables params

### PR DESCRIPTION
**Pull request by dukkee**
Hey @maurerle I think it will be good to have an opportunity to pass regex to the table name filters

### CODE REVIEW

File: eralchemy2/main.py [hunk 1]
```diff
@@ -226,21 +227,27 @@ def filter_resources(
...
     _relationships = copy.deepcopy(relationships)
 
-    include_tables = include_tables or [t.name for t in _tables]
+    include_tables = re.compile(
+        "|".join(f"({name})" for name in (include_tables or [t.name for t in _tables]))
+    )
     include_columns = include_columns or [c.name for t in _tables for c in t.columns]
-    exclude_tables = exclude_tables or list()
+    exclude_tables = re.compile(
+        "|".join(f"({name})" for name in (exclude_tables or [r"\?\?\?"]))
```
Comment (maurerle) [inline eralchemy2/main.py:235]: So a table with the name "???" ist always excluded. This is fine.

File: eralchemy2/main.py [hunk 2]
```diff
@@ -226,21 +227,27 @@ def filter_resources(
...
+        if not exclude_tables.search(t.name) and include_tables.search(t.name)
     ]
     _relationships = [
         r
         for r in _relationships
-        if r.right_col not in exclude_tables
-        and r.left_col not in exclude_tables
-        and r.right_col in include_tables
-        and r.left_col in include_tables
+        if not exclude_tables.search(r.right_col)
```
Comment (maurerle) [inline eralchemy2/main.py:247]: `exclude_table = re.compile("parent")`

This matches also:
`exclude_table.search("parent_table")`

I think we should use [re.fullmatch](https://docs.python.org/3/library/re.html#re.fullmatch) instead, to be consistent with the prior behavior?

**Review by maurerle — COMMENTED** [review]
Hi, thank you for this valuable pull request.

Could you also add tests to `test_filter_exclude_tables`?
Would also be good to have test cases of filters which do not match (like "par" alone without .*) and make sure they work correctly :)

**Comment by maurerle** [comment]
Hi @dukkee - any update on this?

**Comment by dukkee** [comment]
hey @maurerle I'm sorry for so late answer.. Thank you a lot for your help 🙏

**Pull request merged**